pgmpy/models/BayesianModel.py

```diff
@@ -738,6 +747,72 @@ def predict(self, data):
                 pred_values[k].append(v)
         return pd.DataFrame(pred_values, index=data.index)
 
+    def predict_probability(self, data):
```

Comment:
@raghavg7796 I don't like this implementation since most of the code is common in this and predict method. I would suggest you to call `predict_probability` method in `predict`. Or you can also have a new method which has the common part of the code.